app/controllers/users_controller.rb

```diff
@@ -231,6 +231,10 @@ def create
       return fail_with("login.password_too_long")
     end
 
+    if SiteSetting.reserved_usernames.include? params[:username].downcase
+      return fail_with("login.reserved_username")
+    end
+
```

Comment:
Nope, that will add problems for the existing @discourse account created on the hosted sites, and you wouldn't ever be able to add a username to the default list without causing people problems.